dietpi/dietpi-config

```diff
@@ -822,160 +822,441 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
+						tvservice_s_framebuffer_y=${tvservice_s_framebuffer_y##*x}  # retain the part after the last character x
+						tvservice_s_framebuffer_y=${tvservice_s_framebuffer_y::-1} # Remove the last character of blank space
+
+						# default X,Y resolution to framebuffer vars
+						framebuffer_x=$tvservice_s_framebuffer_x
+						framebuffer_y=$tvservice_s_framebuffer_y
+				fi
 
-			local framebuffer_x=$(grep -m1 '^[[:blank:]]*framebuffer_width=' /DietPi/config.txt || vcgencmd get_config framebuffer_width)
-			framebuffer_x=${framebuffer_x#*=}; framebuffer_x=${framebuffer_x:-0}
-			local framebuffer_y=$(grep -m1 '^[[:blank:]]*framebuffer_height=' /DietPi/config.txt || vcgencmd get_config framebuffer_height)
-			framebuffer_y=${framebuffer_y#*=}; framebuffer_y=${framebuffer_y:-0}
-			local current_value=$(grep -m1 '^[[:blank:]]*dtoverlay=vc4-' /DietPi/config.txt | sed 's/^[^=]*=//') # OpenGL check 1st
-			if [[ ! $current_value ]]; then
+				if [[ ! $current_value ]]; then
 
-				# - FB
-				current_value="$framebuffer_x X $framebuffer_y"
+					# - FB
+					current_value="$framebuffer_x X $framebuffer_y"
 
-				# - Check for headless
-				grep -q '^[[:blank:]]*hdmi_ignore_hotplug=1' /DietPi/config.txt && grep -q '^[[:blank:]]*hdmi_ignore_composite=1' /DietPi/config.txt && current_value='Headless'
+					# - check for headless
+					grep -qi 'hdmi_ignore_hotplug=1' /DietPi/config.txt &&
+					grep -qi 'hdmi_ignore_composite=1' /DietPi/config.txt && current_value='Headless'
 
-			fi
+				fi
 
-			G_WHIP_MENU_ARRAY=(
+				#Check if hdmi monitor/tv is detected
+				#Save output of tvservice -m DMT to array by line
+				readarray -t output_of_tvservice_m_DMT_array <<< "$(tvservice -m DMT)"
+				#Save output of tvservice -m CEA to array by line
+				readarray -t output_of_tvservice_m_CEA_array <<< "$(tvservice -m CEA)"
 
-				'vc4-kms-v3d' ': OpenGL | 1920 x 1080'
-				'vc4-fkms-v3d' ': OpenGL | 1920 x 1080'
-				'1080p' ': 1920 x 1080'
-				'720p' ': 1280 x 720'
-				'480p' ': 854 x 480'
-				'RPi Touchscreen' ': 800 x 480'
-				'PC1' ': 1024 x 768'
-				'PC2' ': 800 x 640'
-				'PC3' ': 640 x 480'
-				'DietPi-Cloudshell' ': 320 x 240'
-				'sdtv_mode=0' ': Composite NTSC'
-				'sdtv_mode=1' ': Composite Japanese NTSC'
-				'sdtv_mode=2' ': Composite PAL'
-				'sdtv_mode=3' ': Composite Brazilian PAL'
-				'Headless' ': Disables HDMI & Composite Output'
+				#Monitor?  Get value DMT Modes value of zero if none are supported else any other value than zero it is detecting suported modes
+				DMT_detected=${output_of_tvservice_m_DMT_array[0]:14:1}   #first 1 characters after removing the first 14
+				#TV? Get value CEA Modes value of zero if none are supported else any other value than zero it is detecting suported modes
+				CEA_detected=${output_of_tvservice_m_CEA_array[0]:14:1}   #first 1 characters after removing the first 14
 
-			)
+				if [ "$DMT_detected" != '0' ]] || [[ "$CEA_detected" != '0' ]]; then
```

Comment:
One missing bracket and as above:
`[[ $DMT_detected != '0' || $CEA_detected != '0' ]]` or here arithmetic check makes sense to not list entries in case of invalid value:
`(( $DMT_detected || $CEA_detected ))`